internal/controllers/addon/phase_ensure_catalogsource_test.go

```diff
@@ -76,101 +76,39 @@ func TestReconcileCatalogSource_NotExistingYet_WithClientErrorCreate(t *testing.
 	).Return(timeoutErr)
 
 	ctx := context.Background()
-	_, err := reconcileCatalogSource(ctx, c, testutil.NewTestCatalogSource(), addonsv1alpha1.ResourceAdoptionAdoptAll)
+	_, err := reconcileCatalogSource(ctx, c, testutil.NewTestCatalogSource())
 	assert.Error(t, err)
 	assert.EqualError(t, err, timeoutErr.Error())
 	c.AssertExpectations(t)
 }
 
 func TestReconcileCatalogSource_Adoption(t *testing.T) {
```

Comment:
Is this test method even needed anymore? 

From my pov, it is just essentially doing this

```
assert.True(t, equality.Semantic.DeepEqual(reconciledCatalogSource.OwnerReferences, catalogSource.OwnerReferences))
```
but isn't that already very straightforwardly implied by this piece of code in phase_ensure_catalogsource.go

```go
if specChanged || !ownedByAddon || !labels.Equals(newLabels, currentLabels) {
		currentCatalogSource.Spec = catalogSource.Spec
		currentCatalogSource.OwnerReferences = catalogSource.OwnerReferences
		currentCatalogSource.Labels = newLabels
```
Just want to ensure that we aren't over-testing stuff and cluttering up the test code 😅 

wdyt :)

Reply:
Hey Yash :) Yeah, it is true the logic being tested isn't especially complex and that this is more of a smoke test. Also one good aspect of it might be that is show cases the new default behavior of adopting everything. But either way I don't have a super strong opinion, what do you think @thetechnick ?

Reply:
We test behavior not implementation. Regardless of what the implementation looks like now we should have a regression test like this to protect against future breaking changes (which already happened and is why we added this test). That being said since this is a more general feature now, moving the assertion to a more general test which verifies what happens when a catalog source already exists at the time of reconciliation might make sense.

Reply:
Hmm, that's good point @Ajpantuso , let's go ahead with this then :)